processor/src/main/java/org/mapstruct/ap/internal/model/source/Mapping.java

```diff
@@ -475,9 +475,6 @@ public Mapping copyForForwardInheritance( SourceMethod method ) {
 
     @Override
     public boolean equals(Object o) {
-        if ( this == o ) {
```

Comment:
Why remove this? This makes is a shortcut when you check against the same instance.

Reply:
Not sure whether this is needed still. There was a reason for removing this earlier.. I guess its a left over and I'll remove it again.